dev/pytest

```diff
@@ -38,17 +38,19 @@ if [ -n "$KOALAS_USAGE_LOGGER" ]; then
 fi
 
 # Runs both doctests and unit tests by default, otherwise hands arguments over to pytest.
+# we're doing pytest with `xvfb-run` since without this some of the tests would fail for GUI library because there is no display server.
+# The X virtual framebuffer (Xvfb) display server is used to perform all the graphical operations in memory without showing any screen output.
 if [ "$#" = 0 ]; then
     if [[ "$SPARK_VERSION" == 2.3* ]] || [[ "$SPARK_VERSION" == 2.4.1* ]] || [[ "$SPARK_VERSION" == 2.4.2* ]]; then
         # Delta requires Spark 2.4.2+. We skip the related doctests.
         if [[ "$SPARK_VERSION" == 2.3* ]]; then
-            pytest --cov=databricks -k " -melt -to_delta -read_delta" --verbose --showlocals --doctest-modules databricks "${logopts[@]}"
+            xvfb-run python -m pytest --cov=databricks -k " -melt -to_delta -read_delta" --verbose --showlocals --doctest-modules databricks "${logopts[@]}"
         else
-            pytest --cov=databricks -k "-to_delta -read_delta" --verbose --showlocals --doctest-modules databricks "${logopts[@]}"
+            xvfb-run python -m pytest --cov=databricks -k "-to_delta -read_delta" --verbose --showlocals --doctest-modules databricks "${logopts[@]}"
```

Comment:
@itholic, if we do this, the dev env requires `xvfb-run` but `xvfb-run` is only required when the display isn't available. From the offline discussions, it matters in `to_clipboard` test right? Let's just change the test if it throws the same exception or same output.